### PR TITLE
Use an array to manage index of speed limit by tag entries in SpeedSectionMetadataForm

### LINKED ISSUE #6449: Improve the speed limit form in Editor application
### Description and goal

In the 'Speed Limit by Tag' form, we currently display each tag alongside its value using a pair of input fields. These tags and their corresponding values are stored within a Record data structure. To render these input fields, we iterate over the entries of this Record.

The structure involves a <div> element that wraps around each pair of input fields. For each iteration over the record's entries, we utilize the index from the entries array of the Record as a key for React elements.

**Issue:**

This approach, while functional, has been identified as inefficient. Relying on array indices as keys in React is not optimal, especially when the list can change. This can lead to performance issues and bugs.

**Proposed Solution:**

To enhance efficiency and ensure stable identities for each set of inputs, we propose the introduction of a unique identifier for each tag-value pair. This could involve:

- Generating a unique ID for each tag-value pair when it is created or fetched.
- Using this unique ID as the key instead of the array index.
This change is expected to improve the performance of the form rendering and prevent potential issues related to React's handling of list item reordering or updates.

### Acceptance criteria

- Ensure elements no longer use array indices as keys.
- Validate that editing, adding and deleting speed limits works correctly and updates the user interface appropriately.
- Verify that the data sent to the backend accurately reflects the inputs made in the form.

### PR DESCRIPTION
**Pull request by kmer2016**
closes #6449

### CODE REVIEW

**Review by Yohh — COMMENTED** [review]
lgtm for me, checked, but the build fails in CI

File: front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionMetadataForm.tsx [hunk 1]
```diff
@@ -31,6 +32,17 @@ const SpeedSectionMetadataForm: FC = () => {
...
+  const [uniqueIds, setUniqueIds] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (!isEmpty(entity.properties.speed_limit_by_tag)) {
+      const newUniqIds: string[] = Object.keys(entity.properties.speed_limit_by_tag).map(() =>
+        nextId()
+      );
+      setUniqueIds(newUniqIds);
+    }
+  }, []);
```
Comment (SharglutDev) [inline front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionMetadataForm.tsx]: If we try to edit a speed section who already have some speed limit by tags, we still get at first load the "duplicated key" warning in console. 
Maybe even with the useEffect with an empty dependency array, the first render of the components starts before the state has been updated.

It seems that the solution below fixes the warning.

```suggestion
  const [uniqueIds, setUniqueIds] = useState<string[]>(
    Object.keys(entity.properties?.speed_limit_by_tag || {}).map(() => nextId())
  );
```

**Review by SharglutDev — CHANGES_REQUESTED** [review]
Thank you for your PR. We are on the good path, there is still one thing to fix, see comments.

File: front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionMetadataForm.tsx (hunk 1 above)
Comment (kmer2016) [reply front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionMetadataForm.tsx]: Done !

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested

**Pull request merged**